changelog/dmd.boolean-array-literal.dd

```diff
@@ -0,0 +1,26 @@
+Boolean evaluation of array literals and string literals is now deprecated
+
+Boolean evaluation of a string literal could happen unintentionally
+e.g. when an `assert(0, "message")` was meant and the `0` was missing.
+
+```d
+assert("unexpected runtime condition");
+static assert("unhandled case for `", T, "`");
+
+enum int[] arr = [];
+...
+if (arr) { ... } // was arr.length meant?
```

Comment:
The `assert` change is good but having it for all expression seems like a mistake to me. The line I'm commenting on isn't a good change. I agree it should mean `arr.length` instead of `arr.ptr` but deprecating it won't fix that.

Reply:
Now done: https://github.com/dlang/dmd/pull/15860